packages/php-domain/src/TextRoles/InterfaceTextRole.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3Docs\PhpDomain\TextRoles;
+
+use phpDocumentor\Guides\Nodes\Inline\AbstractLinkInlineNode;
+use phpDocumentor\Guides\Nodes\Inline\ReferenceNode;
+use phpDocumentor\Guides\ReferenceResolvers\AnchorReducer;
+use phpDocumentor\Guides\RestructuredText\Parser\DocumentParserContext;
+use phpDocumentor\Guides\RestructuredText\Parser\InlineLexer;
+use phpDocumentor\Guides\RestructuredText\TextRoles\TextRole;
+use Psr\Log\LoggerInterface;
+
+final class InterfaceTextRole implements TextRole
+{
+    private readonly InlineLexer $lexer;
+    public function __construct(
+        private readonly LoggerInterface $logger,
+        private readonly AnchorReducer $anchorReducer,
+    ) {
+        // Do not inject the $lexer. It contains a state.
+        $this->lexer = new InlineLexer();
+    }
+    final public const NAME = 'interface';
+
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+    public function getAliases(): array
+    {
+        return [];
+    }
+
+    public function processNode(
```

Comment:
should we maybe extract this to a trait in guides to make is reusable? This seems to me a very complex piece of code, and we are reusing this very often.

Reply:
We do have it in a the GeneralLinkTextRole, however we have a special case as all other textroles are escaping the links (\\ -> \) while this text role does not want to escape backslashes. So we cannot reuse the code.